client/js/lounge.js

```diff
@@ -994,6 +1005,22 @@ $(function() {
 		});
 	});
 
+
+	// TODO: doesn't support channels correctly, needs to change %23 to #
+	// 		 boolean not supported, like TLS
+	//       port doesn't work because it has no ID
+
+	// name = network, host = server
+	windows.on("show", "#connect", function() {
+		var url_config_items = ["name", "host", "password", "nick", "username", "realname", "channels"];
+		url_config_items.forEach(function(e) {
+			var key = getQueryParam(e);
```

Comment:
Have your query param extractor return an object to avoid parsing the URL again and again needlessly.